### PR TITLE
[BEAM-3176] support drop table

### PR DESCRIPTION
**Pull request by xumingming**
Support drop table statement in BeamSQL.

---

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by xumingming** [comment]
retest this please

File: sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlCliTest.java [hunk 1]
```diff
@@ -48,6 +49,29 @@ public void testExecute_createTextTable() throws Exception {
     assertNotNull(table);
   }
 
+  @Test
+  public void testExecute_dropTable() throws Exception {
```
Comment (XuMingmin) [inline sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlCliTest.java]: can you add another unit test to see if the table is not available in `Planner` as expected?

**Comment by jkff** [comment]
What are the next steps here?

**Comment by XuMingmin** [comment]
ohhh, this is a PR I don't noticed for long time. 
@xumingming  could you restate the code and I'll verify it again quickly.

**Comment by xumingming** [comment]
Sure, let me rebase it.

**Comment by XuMingmin** [comment]
CI build fails with two errors, can you take a look?
```
2018-02-10T15:08:03.066 [INFO] --- findbugs-maven-plugin:3.0.5:check (default) @ beam-sdks-java-extensions-sql ---
2018-02-10T15:08:03.076 [INFO] BugInstance size is 2
2018-02-10T15:08:03.077 [INFO] Error size is 0
2018-02-10T15:08:03.079 [INFO] Total bugs: 2
2018-02-10T15:08:03.087 [INFO] The field org.apache.beam.sdk.extensions.sql.impl.BeamSqlEnv.tables is transient but isn't set by deserialization [org.apache.beam.sdk.extensions.sql.impl.BeamSqlEnv] In BeamSqlEnv.java SE_TRANSIENT_FIELD_NOT_RESTORED
2018-02-10T15:08:03.087 [INFO] org.apache.beam.sdk.extensions.sql.impl.BeamSqlEnv.deregisterTable(String) makes inefficient use of keySet iterator instead of entrySet iterator [org.apache.beam.sdk.extensions.sql.impl.BeamSqlEnv] At BeamSqlEnv.java:[line 102] WMI_WRONG_MAP_ITERATOR
```

**Comment by XuMingmin** [comment]
retest this please

**Pull request merged**